### ISSUE TITLE
ASP.NET Core Blazor Language Services 15.7.10222 crash VS

### ISSUE THREAD

**Issue by aguacongas** [aspnet/Blazor#597]
I had VS crashes after updating Blazor extension  

> An unhandled exception of type 'System.InvalidOperationException' occurred in Microsoft.AspNetCore.Blazor.Razor.Extensions.dll
> 
> Additional information: Attribute nodes should either be minimized or a single content node.

**Comment by SIkebe** [comment]
@aguacongas Output `dotnet --info`?
Did you install .NET Core SDK 2.1.300-preview2-008533 ?

**Comment by SteveSandersonMS** [comment]
@rynowak This does repro. After creating the minimal `UserName.cshtml` for the above code to be valid, performing that edit (then alt-tabbing out of VS then back in, for some reason) made VS hang and eventually get shut down by the OS. Is this something you could investigate when convenient?

**Comment by floreseken** [comment]
I'm having also troubles loading my blazor solution, this was working fine with 0.1.
VS keeps saying "Repairing solution" just forever. If I exclude my blazor project I loads fine.

Separate (without the solution) the blazor project loads fine.
So only the combination fails.

**Comment by aguacongas** [comment]
@floreseken delete .vs folder, it should fix it

**Comment by rynowak** [comment]
I unfortunately won't have access to a windows machine until Saturday. I'm at a conference and mine blew up. I'll try to write a test that reproduces the issue

**Comment by floreseken** [comment]
Thanks @aguacongas ! that worked.

**Comment by SteveSandersonMS** [comment]
@rynowak No worries - not urgent.

**Comment by rynowak** [comment]
I haven't been able to write a unit test that reproduces the problem.

The exception is here: https://github.com/aspnet/Blazor/blob/dev/src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorDesignTimeNodeWriter.cs#L424

My bet is that something in the bind lowering phase is rewriting this node incorrectly.

The reason why tabbing back and away causes the crash is that VS uses that as a trigger to force the generated code to update. Normal edits don't trigger a full code generation pass, they attempt to reuse and modify the existing codegen where possible.

**Comment by EdCharbeneau** [comment]
I'm having the same issue. I have per the example in the Blazor 0.2.0 release notes on MSDN.

When I type `bind-Content` on my component, VS instantly freezes and crashes with the error: `ASP.NET Core Blazor Language Services 15.7.10222 ....`

Component code (reduced for clarity)
```
   public string Content { get; set; }
    public string FromUrl { get; set; }
    public Action<string> ContentChanged { get; set; }
```
Index.cshtml
```
<MyComponent bind-Content="@ContentValue"/>

@functions {
 public string ContentValue { get; set; } = "Let's do it";
}
```

**Comment by danroth27** [comment]
I hit this a couple of times as well when investigating https://github.com/aspnet/Blazor/issues/610

**Comment by borakasmer** [comment]
Hi!
I install .NET Core SDK 2.1.300-preview2-008533 on my Mac (Os x- macOS High Sierra). And Blazor Template. But when I create new blazer project always comes to old version "0.1.0"!

**Comment by rynowak** [comment]
We've fixed the problem in the compiler. Keeping this open for now since folks with the 0.2.0 tooling will still hit it.

**Comment by SteveSandersonMS** [comment]
@rynowak Your fix for that is bc786e0253947dea35e5984531b026d446cef3ec, right? I'm going to cherry-pick that over into a new `release/0.2.1` branch (taken from `release/0.2.0`). Do you believe that's the only commit needed to fix the crashing issue, and the only thing we should release as 0.2.1? Dan and I would like to put this out as soon as we can.

**Comment by rynowak** [comment]
Yeah, that's what I had in mind. Sounds good

**Comment by SteveSandersonMS** [comment]
The fix for this is now published in the 0.2.1 release.

### How to update

You only need to update your Visual Studio extension. In VS, go to *Tools*, then *Extensions and Updates*, then in *Updates* select the *Visual Studio Marketplace* item. It should prompt you to update the *ASP.NET Core Blazor Language Services* extension to version 15.7.10237 (or later, depending on when you're reading this). Update to the latest version.

You don't need to update your project to the newly-released 0.2.1 package versions. You can do if you want to, but it makes no difference as they are identical to the 0.2.0 packages. The only reason we published updated 0.2.1 packages is because newly-created projects will reference 0.2.1 (our build system updates this automatically).
